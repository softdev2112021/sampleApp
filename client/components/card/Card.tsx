import React from "react";
import CardDetails from './cardDetails/CardDetails';

const Card = ({ id, title, date: { weekDay, date }, content: { data, descr, icon }, contentDetails, onDelete, }) => {
  const cardDetails = contentDetails.map((content) => {
    return <CardDetails key={content.date.weekDay} {...content}/>
  });

  return (
    <div className="col-sm-6 col-md-4 col-xl-3">
      <div className="panel panel-inverse">
        <div className="panel-heading bg-dark text-grey">
          <h4 className="panel-title">
            <b className="f-s-16">{title}</b>
          </h4>
          <div className="panel-date m-r-20">
            <b>{weekDay}/</b>
            <b>{date}</b>
          </div>
          <div className="panel-heading-btn">
            <button
              className="btn btn-xs btn-icon btn-circle btn-danger"
              onClick={() => onDelete(id)}
            >
              <i className="fa fa-times"></i>
            </button>
          </div>
        </div>
        <div className="panel-body pt-0 pb-0">
          <div className="d-flex align-items-center">
            <h2 className="mb-0">
              {data}
              <sup>o</sup>C
            </h2>
            <img src={icon} alt="image" width="64" />
            <b>{descr}</b>
          </div>
        </div>
        <div className="row row-space-0 p-t-10 p-b-10 bg-gradient-grey">{cardDetails}</div>
      </div>
    </div>
  );
};

export default Card;
