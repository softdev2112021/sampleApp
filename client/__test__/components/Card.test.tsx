import React from 'react';
import renderer from 'react-test-renderer';
import Card, { CardProps } from '../../components/card/Card';

it('renders correctly', () => {
  const cardProps: CardProps = {
    id: 1,
    title: "Dnipro",
    date: { weekDay: "Mon", date: "17/Mar" },
    content: {
      data: 20,
      descr: "descr",
      icon: "icon",
    },
    contentDetails: [
      {
        key: "key",
        date: { weekDay: "Mon", date: "17/Mar" },
        content: {
          data: { min: 19, max: 20, pop: "0%" },
          descr: "descr",
          icon: "icon",
        },
      },
    ],
    onDelete: () => {},
  };

  const tree = renderer
    .create(<Card {...cardProps}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});