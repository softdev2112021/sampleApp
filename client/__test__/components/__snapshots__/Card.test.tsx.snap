// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-sm-6 col-md-4 col-xl-3"
>
  <div
    className="panel panel-inverse"
  >
    <div
      className="panel-heading bg-dark text-grey"
    >
      <h4
        className="panel-title"
      >
        <b
          className="f-s-16"
        >
          Dnipro
        </b>
      </h4>
      <div
        className="panel-date m-r-20"
      >
        <b>
          Mon
          /
        </b>
        <b>
          17/Mar
        </b>
      </div>
      <div
        className="panel-heading-btn"
      >
        <button
          className="btn btn-xs btn-icon btn-circle btn-danger"
          onClick={[Function]}
        >
          <i
            className="fa fa-times"
          />
        </button>
      </div>
    </div>
    <div
      className="panel-body pt-0 pb-0"
    >
      <div
        className="d-flex align-items-center"
      >
        <h2
          className="mb-0"
        >
          20
          <sup>
            o
          </sup>
          C
        </h2>
        <img
          alt="image"
          src="icon"
          width="64"
        />
        <b>
          descr
        </b>
      </div>
    </div>
    <div
      className="row row-space-0 p-t-10 p-b-10 bg-gradient-grey"
    >
      <div
        className="col-4 mb-2 d-flex flex-column align-items-center"
      >
        <div
          className="f-s-12 f-w-800 height-10"
        >
          Mon
          /
          17/Mar
        </div>
        <div
          className="height-50"
        >
          <img
            alt="image"
            src="icon"
            width="64"
          />
        </div>
        <div
          className="f-s-18 f-w-800 height-20"
        >
          20
          <sup>
            o
          </sup>
          C
        </div>
        <div
          className="f-s-14 f-w-800 mb-2"
        >
          19
        </div>
        <i
          className="fa fa-umbrella"
        >
           0%
        </i>
      </div>
    </div>
  </div>
</div>
`;
